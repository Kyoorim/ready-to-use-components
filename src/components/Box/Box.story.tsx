import React, { FC } from 'react';
import { Meta } from '@storybook/react';
import Box, { BoxProps } from './Box';

export default {
  title: 'Box',
  component: Box,
} as Meta<BoxProps>;

// title: 컴포넌트 이름. '/'를 넣어 카테고리화 가능
// component: 어떤 컴포넌트를 가져와서 스토리를 만들 것인지 명시

export const Default: FC = () => {
  return <Box>hello world</Box>;
};
