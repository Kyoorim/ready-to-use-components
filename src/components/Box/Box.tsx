import styled from 'styled-components';
import {
  space,
  layout,
  color,
  flexbox,
  SpaceProps,
  LayoutProps,
  ColorProps,
  FlexboxProps,
  typography,
  TypographyProps,
} from 'styled-system';
import React from 'react';

const StyledBox = styled.div`
  ${space}
  ${layout}
  ${color}
  ${flexbox}
  ${typography}
`;

export type BoxProps = SpaceProps & LayoutProps & ColorProps & FlexboxProps & TypographyProps;

const Box: FCC<BoxProps> = (props: any) => {
  return <StyledBox {...props} data-testid={props['data-testid'] ? props['data-testid'] : 'box-comp'} />;
};

export default Box;
