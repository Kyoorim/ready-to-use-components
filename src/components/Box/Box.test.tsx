import React from 'react';
import { render, screen } from '@testing-library/react';
import Box from './Box';

const defaultProps = {
  onClick: jest.fn(),
  children: 'Box',
};

describe('<Box />', () => {
  it('should render', () => {
    render(<Box {...defaultProps} />);

    expect(screen.getByTestId('box-comp')).toBeInTheDocument();
  });

  it('should render children', () => {
    render(<Box {...defaultProps} />);

    expect(screen.getByText(/Box/i)).toBeInTheDocument();
  });

  it('should allow custom className', () => {
    const props = {
      ...defaultProps,
      className: 'Custom',
    };
    render(<Box {...props} />);

    expect(screen.getByTestId('box-comp')).toHaveProperty('className');
  });
});
