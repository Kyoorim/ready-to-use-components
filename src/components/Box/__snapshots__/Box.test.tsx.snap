// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render 1`] = `
<div
  class="sc-bcXHqe kFhsOn"
  data-testid="box-comp"
>
  Box
</div>
`;
